src/scenario.rs

```diff
@@ -0,0 +1,586 @@
+	/// - `does_block_monitoring` is set to `false`.
+	/// - `installs_ctrl_c_stop_hook` is set to `false`.
+	/// - `send_threshold` is set to `1000`.
+	pub fn new() -> Self {
+		ScenarioBuilder {
+			account_id: None,
+			start_id: None,
+			last_id: None,
+			nonce_from: None,
+			txs_count: 1,
+			tx_recipe: Some(TransactionRecipe::transfer()),
+			does_block_monitoring: false,
+			watched_txs: false,
+			send_threshold: Some(1000),
+			rpc_uri: None,
+			chain_type: None,
+			installs_ctrl_c_stop_hook: false,
+			log_prefix: None,
+			tip: 0,
+		}
+	}
+
+	/// Configure the account id for building a batch of transactions based on a single signer.
+	/// The setter parameter is a string that can be in the form of a number, in which case it will
+	/// behave the same as using `with_start_id` (without `with_last_id`), but it can receive
+	/// a derivation path like the usual Polkadot development accounts (e.g. "alice", "bob", etc).
+	pub fn with_account_id(mut self, account_id: String) -> Self {
+		self.account_id = Some(account_id);
+		self
+	}
+
+	/// Configure the account id for the first signer used for the transactions building.
+	/// If the builder isn't configured with a last signer account id, then the scenario
+	/// builder will build transactions only for the account specified with this setter.
+	///
+	/// It is usually used in pair with `with_last_id`, to set an ids range where each id will be
+	/// the last part of a derivation path used for multiple accounts generation, each being a
+	/// signer for a batch of transactions.
+	pub fn with_start_id(mut self, start_id: u32) -> Self {
+		self.start_id = Some(start_id);
+		self
+	}
+
+	/// Last id of an account signer that is also representing the end of an ids range,
+	/// each id being the last part of a derivation path used to generate accounts that sign a set
+	/// of transactions (see
+	/// [`crate::subxt_transaction::derive_accounts`]).
+	pub fn with_last_id(mut self, last_id: u32) -> Self {
+		self.last_id = Some(last_id);
+		self
+	}
+
+	/// The start of a nonce counter that's incremented with each built transaction, in relation to
+	/// a specific signer account (that can be also part of a range of accounts as it happens when
+	/// both `start_id` and `last_id` parameters of the builder are set), while the number of the
+	/// built transactions is lower than `txs_count`.
+	pub fn with_nonce_from(mut self, nonce_from: Option<u128>) -> Self {
+		self.nonce_from = nonce_from;
+		self
+	}
+
+	/// The number of the transactions that will be built in relation to a signer account.
+	pub fn with_txs_count(mut self, txs_count: u32) -> Self {
+		self.txs_count = txs_count;
+		self
+	}
+
+	/// The builder is already initialised with a transfer transaction recipe for the built
+	/// transactions, and this API
+	pub fn with_transfer_tx_recipe(mut self) -> Self {
+		let mut recipe = TransactionRecipe::transfer();
+		recipe.tip = self.tip;
+		self.tx_recipe = Some(recipe);
```

Comment:
That would be nicer:
```suggestion
		self.tx_recipe = Some(TransactionRecipe::transfer(self.tip));
```
(or `transfer_with_tip` ?)

Reply:
Yeh, this is better. I think we don't need to store the `tip` on the builder (and have a dedicated setter for it), but instead initialize the builder with a transfer recipe with tip 0, and then if people want to set a certain tip, they would need to set the recipe as: `with_transfer_recipe(mut self, tip: u128)`, or `with_remark_recipe(mut self, tip: u128)`. WDYT?

Reply:
I would keep it in builder. It is a way to control priority of the transaction. There will be one more - mortality - which will also go to the recipe. I feel that having dedicated setters will be easier on both sides - user and builder implementation. Also more recipes may come - eg nft minting or similar.

Reply:
At that time probably a dedicated recipe builder sounds better, but yeh, nesting the recipe building in the scenario is also an option for the time being. Will change it quickly

Reply:
Done here: [b1d48c2](https://github.com/michalkucharczyk/tx-test-tool/pull/22/commits/b1d48c212749f0ef791b963f4e8a90340ce6dfba).